Lib/warnings.py

```diff
@@ -590,17 +590,19 @@ def __call__(self, arg, /):
         if category is None:
             arg.__deprecated__ = msg
             return arg
-        elif isinstance(arg, type):
+
+        if isinstance(arg, type):
             import functools
             from types import MethodType
 
             original_new = arg.__new__
+            is_object_new = original_new is object.__new__
 
             @functools.wraps(original_new)
             def __new__(cls, *args, **kwargs):
```

Comment:
A question might be out of the scope of this PR: Should we make `cls` a positional-only argument?

```suggestion
            def __new__(cls, /, *args, **kwargs):
```

A `'cls'` key will never be collected in `kwargs`.

Reply:
Yes, we probably should. Feel free to send a separate PR just for that.